### ISSUE TITLE
Search and filter features combination does not work

### ISSUE THREAD

**Issue by dana2208** [AllenFang/react-bootstrap-table#329]
Hi,

I think the search feature should search on already filtered records and filter feature should work on the results of the search result.
Today it does not work this way. The last one that is updated (either the search textbox or a filter on a column) is the one taking effect on the results displayed,

Here is a code where you can test both search and filter features on the same table:
```
'use strict';
import React from 'react';
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';

function getProducts() {
    const products = [];
    const startId = products.length;
    for (var i = 0; i < 12; i++) {
        var id = startId + i;
        products.push({
            id: id,
            name: "Item name " + id,
            price: Math.floor((Math.random() * 2000) + 1)
        });
    }
    return products;
}

export default class Test extends React.Component {
    constructor(props) {
        super(props);
        this.products = getProducts();
        this.state = {
            data: this.products
        }
    }

    render() {
        return (
            <BootstrapTable data={this.state.data} search={true} multiColumnSearch={true}>
                <TableHeaderColumn dataField="id" isKey={true}>Product ID</TableHeaderColumn>
                <TableHeaderColumn dataField="name" filter={{type: "TextFilter"}}>Product Name</TableHeaderColumn>
                <TableHeaderColumn dataField="price" filter={{type: "NumberFilter", numberComparators: ["=", ">", "<="]}}>Product Price</TableHeaderColumn>
            </BootstrapTable>
        );
    }
};
```

**Comment by AllenFang** [comment]
This seems like a bug. I'll fix it after **```v2.0.0```**

**Comment by AllenFang** [comment]
Fixed on ```v2.0.1```, check it out :)
